### ISSUE TITLE
[ldd-rings] Update Rings dictionary for R&A F ring mosaics project.

### ISSUE THREAD

**Issue by mace-space** [pds-data-dictionaries/PDS4-LDD-Issue-Repo#301]
The following edits/improvements need to be implemented in the Rings dictionary, as discussed with the data provider (Rob French) and RMS team for reprojected calibrated ISS mosaics of Saturn's F ring. 

1. `minimum_incidence_angle` and `maximum_incidence_angle` need to be added to the `Reprojection_Geometry` class (via DD_Association).

2. The following reproduction grid attributes need to be included, but whether this will be in the Rings or CART LDD remains to be discussed: `reprojection_grid_radial_resolution_mean`, `reprojection_grid_radial_resolution_min`, `reprojection_grid_radial_resolution_max`, `reprojection_grid_angular_resolution_mean`, `reprojection_grid_angular_resolution_min`, `reprojection_grid_angular_resolution_max`.

3. For `emission_angle`, the definition says "values in the range of 90 to 180 degrees are possible for ring data" but it provides no information on when the emission angle should be 0-90 and when it should be 90-180. Historically, emission angles for rings are 0-90 on the lit side of the rings, and 90-180 on the unlit side, but that isn't mentioned. Also, the final comma should be a semi-colon, or a period. 

4. The definition for `light_source_incidence_angle` would make sense except for the statement "The value is always equal to 90 - | observed_ring_elevation |”. This is incorrect as observed_ring_elevation is a statement of the angle of the rings with the observer, while `light_source_incidence_angle` is a statement of the angle of the rings with the light source and so they are not related in any way, *except during the one case of stellar occultations, when this holds true*. Therefore, add a statement to this effect in the definition of `light_source_incidence_angle`.

5. Note that for above point 4., `light_source_incidence_angle` is *not* actually being used in the F ring mosaic labels but it came up during discussion for the project and its definition could be improved alongside the other updates. 

These are to be implemented against PDS4 IM Version 1M00 (current latest build).

**Comment by rfrenchseti** [comment]
Going through my label template and the `PDS4_RINGS_1K00_1C00.xsd` file, I see the following additional issues related to my project:

- The definition for `corotating_flag` appears to have a cut-and-paste error. It says:

```
      <xs:documentation>The corotating_flag is a yes-or-no flag that
        indicates whether or not a corotating frame is used. a table
        field. </xs:documentation>
```

- There is inconsistency in the documentation's use of "The" and "element". For example:

```
      <xs:documentation>The mean_phase_angle element provides the mean
        value during the observation for the phase angle at the target.
        </xs:documentation>

      <xs:documentation>mean_observed_ring_elevation specifes the mean
        value for observed_ring_elevation in the data file.
        </xs:documentation>
```

The latter (without "The") is more common.

- "specifies" is misspelled as "specifes` in many definitions.

- Many elements such as `mean_phase_angle` and `minimum_incidence_angle` have the wording "during the observation...at the target". Perhaps this makes more sense for an occultation, which is a point in space but spread out over time. But in the case of an image, I would expect something more like "for the target in the field of view at the midtime of the observation".

- Add `mean_` versions of `minimum_`/`maximum_` lighting geometry fields that don't have them, like `mean_incidence_angle` and `mean_observed_ring_elevation`.

- None of the `phase_angle` definitions actually say what the phase angle means.

- The definition of `incidence_angle` is unhelpful. It doesn't say what the reference is for 0-90 or 90-180, just that "values in the range of 90 to 180 degrees are possible for ring data." (This is the rings dictionary - I wouldn't think we would need to say this).

- The definition of `observed_ring_elevation` is pretty good, but is still occultation-specific.

- The is a separate element for `inertial_ring_longitude` that is mandatory in occultation tables. Should there also be a separate element for `corotating_ring_longitude`, in addition to the min/max versions? It doesn't seem parallel at the moment.

- The definitions for `minimum_ring_radius` and `maximum_ring_radius` are occultation-specific:

```
      <xs:documentation>minimum_ring_radius indicates the smallest ring
        radius value in the data table. Units are km and are always
        positive. Required in label files for ring occultation data.
        </xs:documentation>
```
- As previously discussed, I strongly feel that the new reprojection grid attributes should be in the Rings LDD. CART reprojection is a different animal.

**Comment by matthewtiscareno** [comment]
@rfrenchseti:  We made more progress on these items today.  Here are some responses:

> 1. `minimum_incidence_angle` and `maximum_incidence_angle` need to be added to the `Reprojection_Geometry` class (via DD_Association).

Not only this, but we need to do a lot of reorganizing the flat structure of attributes within classes to use a lot more subclasses.  We have been discussing this and we hope to start implementing it next month. 

> 2\. The following reproduction grid attributes need to be included, but whether this will be in the Rings or CART LDD remains to be discussed: `reprojection_grid_radial_resolution_mean`, `reprojection_grid_radial_resolution_min`, `reprojection_grid_radial_resolution_max`, `reprojection_grid_angular_resolution_mean`, `reprojection_grid_angular_resolution_min`, `reprojection_grid_angular_resolution_max`.

These have been added to the IngestLDD file along with a class to contain them.  However, we need to draft definitions for them (perhaps you can help?) and we also need to create unit tests for them.  The latter will wait until we have completed the reorganization mentioned above. 

> * For `emission_angle`, the definition says "values in the range of 90 to 180 degrees are possible for ring data" but it provides no information on when the emission angle should be 0-90 and when it should be 90-180. Historically, emission angles for rings are 0-90 on the lit side of the rings, and 90-180 on the unlit side, but that isn't mentioned. Also, the final comma should be a semi-colon, or a period.

This is now fixed in the IngestLDD. 

> * The definition for `light_source_incidence_angle` would make sense except for the statement "The value is always equal to 90 - | observed_ring_elevation |”. This is incorrect as observed_ring_elevation is a statement of the angle of the rings with the observer, while `light_source_incidence_angle` is a statement of the angle of the rings with the light source and so they are not related in any way, _except during the one case of stellar occultations, when this holds true_. Therefore, add a statement to this effect in the definition of `light_source_incidence_angle`.

This is now fixed in the IngestLDD. 

> * Note that for above point 4., `light_source_incidence_angle` is _not_ actually being used in the F ring mosaic labels but it came up during discussion for the project and its definition could be improved alongside the other updates.

Noted.

> * The definition for `corotating_flag` appears to have a cut-and-paste error.

We removed the phrase "a table field"

> * There is inconsistency in the documentation's use of "The" and "element".

Acknowledged, but maybe not worth the trouble to fix.  We have not decided yet.

> * "specifies" is misspelled as "specifes` in many definitions.

This is now fixed in the IngestLDD. 

> * Many elements such as `mean_phase_angle` and `minimum_incidence_angle` have the wording "during the observation...at the target". Perhaps this makes more sense for an occultation, which is a point in space but spread out over time. But in the case of an image, I would expect something more like "for the target in the field of view at the midtime of the observation".

After some consideration, we think that the current language is accurate.  Let's discuss in person sometime. 

> * None of the `phase_angle` definitions actually say what the phase angle means.

The attribute `phase_angle` does seem to have an adequate definition.  Let's discuss in person sometime.

There are more items in this issue not mentioned here.  We will continue our work in the new year.

**Comment by rgdeen** [comment]
Incidence/emission/phase angles are not specific to rings, they're standard photometric properties.  Should this be made more general in some way?  A Photometry class in IMG perhaps, or even a photom: LDD?  I imagine what *is* ring-specific here is specifying the context... specifying what exactly you are measuring the IEP of.  Could be a great use case for cross-dictionary reuse... reuse a class like img:Photometry in the Rings LDD such that the context is clear.

**Comment by matthewtiscareno** [comment]
@rgdeen:  That's correct.  The photometric angles in the Rings LDD are currently duplicates of similar attributes in the Geometry LDD.  In fact, I just made an [adjustment](https://github.com/pds-data-dictionaries/ldd-geom/pull/57) in the latter to correspond to an adjustment we're making here.  

If we resolve [CCB#41](https://github.com/NASA-PDS/PDS4-CCB/issues/41), then we might unify these duplicated instances so that each appears only once in the code.

**Comment by rgdeen** [comment]
Why is it dependent on #41?  That is just a validation/enforcement mechanism.  Lack of such a mechanism should not be an excuse to do bad things to the IM.  I'd much rather have a clean IM and deal with enforcement through other means.

If the mechanism I'm proposing for cross-dictionary validation (tickets coming shortly) is approved then you could use that to have a rule in Rings saying whatever this class is can't be  at the Discipline or Mission top-level... which effectively gives you #41 at least for your domain.  For that matter, we could probably put such a rule in Img itself (assuming this class goes in Img) now without any action on #41 or my proposal.  Basically just an assert saying throw an error if either Mission_Area.Photometry or Discipline_Area.Photometry exists (or whatever the class name is).

**Comment by matthewtiscareno** [comment]
The duplication already exists, and has for years.  So we're not "doing bad things to the IM" but only making small adjustments to what already exists. 

The basis of [CCB#41](https://github.com/NASA-PDS/PDS4-CCB/issues/41) is solving the problem of allowing elements to be used in other LDDs without allowing them to be used from top-level elements.  If you have a solution to that problem that is different from the current proposed solution at [CCB#41](https://github.com/NASA-PDS/PDS4-CCB/issues/41), then you should not file a new ticket but rather we should reach consensus and then possibly swap out a new solution to [CCB#41](https://github.com/NASA-PDS/PDS4-CCB/issues/41).

**Comment by matthewtiscareno** [comment]
By the way, you can't just write #41, because GitHub will assume you mean issue #41 in the current repo, which is not what you mean.  Follow the link in my previous sentence (or in your most recent post), and you'll see what I mean.  This is why I use the construction `[CCB#41](https://github.com/NASA-PDS/PDS4-CCB/issues/41)`, which appears as [CCB#41](https://github.com/NASA-PDS/PDS4-CCB/issues/41).

**Comment by rgdeen** [comment]
ok good to know thanks.  I guess that works over on CCB but I forgot this is in the rings space, sorry.  Anyway, was there a proposed solution in [CCB#41](https://github.com/NASA-PDS/PDS4-CCB/issues/41) ?  From what I recall, and at a glance I don't see anything different, there wasn't a proposed solution, just a desire for one.

As I said, I don't think cleanup should be contingent on that.  The mechanism works now, without enforcement, but that's perfectly ok in my book.  However, if this is just tweaks then that cleanup probably belongs in a separate ticket.

**Comment by matthewtiscareno** [comment]
I'm not opposed to getting started with cross-referencing.  It would require exposing things over at Geometry LDD, and then referencing them at Rings LDD.  It may not be part of our current round of changes, but we'll see.

**Comment by matthewtiscareno** [comment]
Hey @rfrenchseti:  @esimpsons3ti is returning to this task.  We previously added the class `Reprojection_Grid_Parameters` to the Rings LDD (that is, it is in a draft upgrade that is not yet the current version).  This class contains new attributes `reprojection_grid_radial_resolution_mean`, `reprojection_grid_radial_resolution_min`, `reprojection_grid_radial_resolution_max`, `reprojection_grid_angular_resolution_mean`, `reprojection_grid_angular_resolution_min`, and `reprojection_grid_angular_resolution_max`.  

@esimpsons3ti points out that these attributes currently lack definitions.  Do you have these written, or would you like to help us write them?

**Comment by matthewtiscareno** [comment]
Here is a draft.  Please note that I'm naming these attributes a bit differently than you requested.  I think my first entry might accomplish the purpose you have in mind with your "mean," but I'm willing to discuss it if you think otherwise.  Also, I believe it's a PDS4 principle that the mean/minimum/maximum should be spelled out and should come at the beginning, and it's probably our oversight that we didn't previously have it that way.

> reprojection_grid_[radial/angular]\_resolution indicates the [radial/azimuthal] spacing between consecutive [rows/columns] in a ring reprojected image. In practice, this may be somewhat smaller than the [radial_resolution/angular_resolution] because a profile may be over-sampled.
> Required in labels if the value is fixed. If the value varies, the corresponding minimum and maximum attributes must be used in labels and reprojection_grid_[radial/angular]_resolution should be a table field.

> [minimum/maximum]\_reprojection_grid_[radial/angular]\_resolution indicates the [smallest/largest] [radial/azimuthal] spacing between consecutive [rows/columns] in a ring reprojected image. In practice, this may be somewhat smaller than the [radial_resolution/angular_resolution] because a profile may be over-sampled.
> If the value of reprojection_grid_[radial/angular]_resolution varies, the minimum and maximum attributes are required in labels. Not intended to be used as a table field.

**Comment by matthewtiscareno** [comment]
Also, I notice that the Rings LDD contains `radial_resolution` but does not contain `angular_resolution` (or `azimuthal_resolution` or `longitudinal_resolution`, all of which are alternative names for the same thing).  I believe this is because the Rings LDD has primarily been developed for occultations.  Since we are now using it for reprojected images, we probably should add one of these.

**Comment by matthewtiscareno** [comment]
Okay, here are definitions for the corresponding attributes describing the data product, rather than the derived product.  Note that the "radial" ones already exist, but I'm suggesting modifications to them here.

> [radial/azimuthal]\_resolution indicates the nominal [radial/azimuthal] distance over which changes in ring properties can be detected within a data product. Note: this value may be larger than that of radial_sampling_interval or reprojection_grid_[radial/azimuthal]\_resolution, because a derived product can be over-sampled.
> Required in labels if the value is fixed.  If the value varies, the corresponding minimum and maximum attributes must be used in labels and [radial/azimuthal]_resolution should be a table field.

> [minimum/maximum]\_[radial/azimuthal]\_resolution indicates the [smallest/largest] [radial/azimuthal] distance over which changes in ring properties can be detected within a data product. Note: this value may be larger than that of radial_sampling_interval or reprojection_grid_[radial/azimuthal]\_resolution, because a derived product can be over-sampled.
> If the value of [radial/azimuthal]_resolution varies, the minimum and maximum attributes are required in labels. Not intended to be used as a table field.

**Comment by matthewtiscareno** [comment]
After discussion at the team meeting (specifically after input from Mark), "longitudinal" is better than "azimuthal" when we're talking about the rings coordinate system with origin at the center of Saturn.  Thus, I have edited my comments above to use "longitudinal" instead of "azimuthal" at every point.

**Comment by rfrenchseti** [comment]
A few notes:

- "In practice, this may be somewhat smaller than the [radial_resolution/longitudinal_resolution] because a profile may be over-sampled."

If you're going to include a statement like this, it should say "This will usually be different from and can be either larger or smaller than the [radial_resolution/longitudinal_resolution] because an image may be under- or over-sampled."

First of all, "somewhat" doesn't really add anything quantitatively. Second, there's no reason to believe that a reprojection will over-sample the actual image. Some images are extremely high resolution but we don't want the reprojection to actually represent that because the resulting reprojected image would be too large. Third, we aren't talking about profiles, we're talking about images with a reprojection grid.

- "If the value varies, the corresponding minimum and maximum attributes must be used in labels and reprojection_grid_[radial/longitudinal]_resolution should be a table field."

This is kind of a weird thing to support. Do we really want reprojection grids where the grid isn't uniform?

- "[radial/longitudinal]_resolution indicates the nominal [radial/longitudinal] distance over which changes in ring properties can be detected within a data product. Note: this value may be larger than that of radial_sampling_interval or reprojection_grid_[radial/longitudinal]_resolution, because a derived product can be over-sampled."

Same comment as above about being over-sampled.

But more than this, I don't know what it means to say "distance over which changes in ring properties can be detected". That sounds really subjective, or dependent on the type of image post-processing you do. For example, with a camera with a large PSF, maybe it takes several pixels to "detect a change", but that's not what we're trying to say here. It should be expressed in terms of the gradient of the radial resolution field from one pixel to the next. Same for longitude.

**Comment by matthewtiscareno** [comment]
> If you're going to include a statement like this, it should say "This will usually be different from and can be either larger or smaller than the [radial_resolution/longitudinal_resolution] because an image may be under- or over-sampled."

Good edit (at both instances).  I habitually oversample my reprojections, but you're right that there is a use case for undersampling. 

> Do we really want reprojection grids where the grid isn't uniform?

I suppose you're right.  We can always add this if a use case appears.  I added this without much thought, as an analogy to similar attributes that pertain to occultations.  In an occultation, the data as taken have variable resolution, which is sometimes encoded in this way.  An image also has variable resolution, but that would have to be a backplane and it wouldn't be reprojected. 

> But more than this, I don't know what it means to say "distance over which changes in ring properties can be detected". That sounds really subjective, or dependent on the type of image post-processing you do. For example, with a camera with a large PSF, maybe it takes several pixels to "detect a change", but that's not what we're trying to say here. It should be expressed in terms of the gradient of the radial resolution field from one pixel to the next. Same for longitude.

This language mirrors definitions for attributes that already exist.  If we are going to improve our definition of "resolution," then we should do it across the board.  I'm open to suggestions.

@esimpsons3ti, please take note of all this.

**Comment by rfrenchseti** [comment]
The XML template that @mace-space originally gave me has these fields:

```
<rings:reprojection_grid_radial_resolution_mean>$MEAN_REPROJ_GRID_RAD_RES$</rings:reprojection_grid_radial_resolution_mean>
<rings:reprojection_grid_radial_resolution_min>$MIN_REPROJ_GRID_RAD_RES$</rings:reprojection_grid_radial_resolution_min>
<rings:reprojection_grid_radial_resolution_max>$MAX_REPROJ_GRID_RAD_RES$</rings:reprojection_grid_radial_resolution_max>
<rings:reprojection_grid_angular_resolution_mean>$MEAN_REPROJ_GRID_ANG_RES$</rings:reprojection_grid_radial_resolution_mean>
<rings:reprojection_grid_angular_resolution_min>$MIN_REPROJ_GRID_ANG_RES$</rings:reprojection_grid_radial_resolution_min>
<rings:reprojection_grid_angular_resolution_max>$MAX_REPROJ_GRID_ANG_RES$</rings:reprojection_grid_radial_resolution_max>
```

However, these were being used in a very different way. This was the radial/angular resolution *of the original image* before reprojection, not the size of the reprojection grid itself. In retrospect, the actual resolution of the grid was missing entirely from the template.

One of my data products is a table that, for each reprojected longitude (row in the table), tabulates the minimum, maximum, and mean emission angle, phase angle, radial resolution, and angular resolution for all available radii at that longitude. The question is whether the label should include the min/max/mean values for the entire reprojection, since there are no current dictionary entries to cover that. Honestly I'm going to say at this point that these should not be in the label, and they can be derived by the user from the table if necessary.

The `rings:reprojection_grid_radial_resolution` should be the resolution of the grid itself, and have nothing to do with the resolution of the source images.

**Comment by esimpsons3ti** [comment]
@rfrenchseti so, would this mean that you want the `[maximum/minimum/mean]_reprojection_grid_[radial/longitudinal]_resolution` attributes to be removed from the dictionary entirely? Or keep them in the dictionary for now but remove them from the template?

**Comment by esimpsons3ti** [comment]
I clicked the wrong button! My bad.

**Comment by rfrenchseti** [comment]
> [@rfrenchseti](https://github.com/rfrenchseti) so, would this mean that you want the `[maximum/minimum/mean]_reprojection_grid_[radial/longitudinal]_resolution` attributes to be removed from the dictionary entirely? Or keep them in the dictionary for now but remove them from the template?

Well it seems logical to remove them entirely, right? Is there any case where someone would reproject onto a non-uniform grid?

**Comment by matthewtiscareno** [comment]
> > [@rfrenchseti](https://github.com/rfrenchseti) so, would this mean that you want the `[maximum/minimum/mean]_reprojection_grid_[radial/longitudinal]_resolution` attributes to be removed from the dictionary entirely? Or keep them in the dictionary for now but remove them from the template?
> 
> Well it seems logical to remove them entirely, right? Is there any case where someone would reproject onto a non-uniform grid?

This is fine. 

Just to say, you were the one who had originally requested these, right?  So you're reconsidering your request, with good reasons given.

**Comment by matthewtiscareno** [comment]
Okay, here are updated definitions.  Note that some of the "radial" ones already exist, but I'm suggesting modifications to them here.

I've changed "distance over which changes in ring properties can be detected" ==> "size of a single pixel."  It turns out that all the attributes in which this language already occurs in the Rings LDD are listed below. 

I've restored mean_[radial/longitudinal]\_resolution, which I had overzealously deleted.  I suppose people might want to search on the mean value, while also recording the minimum and maximum. 

In reprojection_grid_[radial/longitudinal]_resolution, I changed "[rows/columns]" ==> "rows or columns" in order to make no assumption about how data providers might rotate their grid.

> [radial/longitudinal]\_resolution indicates the nominal [radial/longitudinal] size of a single pixel. Note: this value may be either smaller or larger than that of radial_sampling_interval or reprojection_grid_[radial/longitudinal]\_resolution, because a derived product can be over-sampled or under-sampled.
> Required in labels if the value is fixed.  If the value varies, the corresponding mean and minimum and maximum attributes must be used in labels and [radial/longitudinal]\_resolution should be a table field.
 
> [mean/minimum/maximum]\_[radial/longitudinal]\_resolution indicates the [mean/smallest/largest] [radial/longitudinal] size of a single pixel. Note: this value may be either smaller or larger than that of radial_sampling_interval or reprojection_grid_[radial/longitudinal]\_resolution, because a derived product can be over-sampled or under-sampled.
> Required in labels if the value of [radial/longitudinal]\_resolution varies. Not intended to be used as a table field.

> reprojection_grid_[radial/longitudinal]_resolution indicates the [radial/longitudinal] size of a single pixel. In practice, this may be either larger or smaller than the [radial/longitudinal]\_resolution because a profile may be over-sampled or under-sampled.
> Required in labels for reprojected images, which are expected to have a fixed grid and thus a constant value for this attribute.

**Comment by matthewtiscareno** [comment]
Okay, here are updated definitions.  Note that some of the "radial" ones already exist, but I'm suggesting modifications to them here.

I've changed "distance over which changes in ring properties can be detected" ==> "size of a single pixel."  It turns out that all the attributes in which this language already occurs in the Rings LDD are listed below. 

I've restored mean_[radial/longitudinal]\_resolution, which I had overzealously deleted.  I suppose people might want to search on the mean value, while also recording the minimum and maximum. 

In reprojection_grid_[radial/longitudinal]_resolution, I changed "[rows/columns]" ==> "rows or columns" in order to make no assumption about how data providers might rotate their grid.

> [radial/longitudinal]\_resolution indicates the nominal [radial/longitudinal] size of a single pixel. Note: this value may be either smaller or larger than that of radial_sampling_interval or reprojection_grid_[radial/longitudinal]\_resolution, because a derived product can be over-sampled or under-sampled.
> Required in labels if the value is fixed.  If the value varies, the corresponding mean and minimum and maximum attributes must be used in labels and [radial/longitudinal]\_resolution should be a table field.
 
> [mean/minimum/maximum]\_[radial/longitudinal]\_resolution indicates the [mean/smallest/largest] [radial/longitudinal] size of a single pixel. Note: this value may be either smaller or larger than that of radial_sampling_interval or reprojection_grid_[radial/longitudinal]\_resolution, because a derived product can be over-sampled or under-sampled.
> Required in labels if the value of [radial/longitudinal]\_resolution varies. Not intended to be used as a table field.

> reprojection_grid_[radial/longitudinal]_sampling_interval indicates the [radial/longitudinal] size of a single pixel in a reprojected image (a derived product). In practice, this may be either larger or smaller than the [radial/longitudinal]\_resolution because a profile or reprojection may be over-sampled or under-sampled.
> Required in labels for reprojected images, which are expected to have a fixed grid and thus a constant value for this attribute.

Note that radial_sampling_interval and [minimum/maximum]\_radial_sampling_interval already exist and are intended to do for occultations what reprojection_grid_[radial/longitudinal]_sampling_interval do for reprojected images.  In fact, it might be nice to change radial_sampling_interval to occultation_radial_sampling_interval, but that would be backwards-incompatible.

**Comment by rfrenchseti** [comment]
@matthewtiscareno Can you just give the complete list of all the attributes (both reprojection grid and other) in one place so that it's easier to review them?

**Comment by matthewtiscareno** [comment]
I believe my message immediately above is a complete list.  Do you see something missing?  Are you saying that you want me to expand out all the combinatorics?  That is, for example, separately give `mean_radial_resolution`, `minimum_radial_resolution`, `maximum_radial_resolution`, `mean_longitudinal_resolution`, `minimum_longitudinal_resolution`, and `maximum_longitudinal_resolution` instead of combining them into `[mean/minimum/maximum]_[radial/longitudinal]_resolution`?

**Comment by rfrenchseti** [comment]
> I believe my message immediately above is a complete list. Do you see something missing? Are you saying that you want me to expand out all the combinatorics? That is, for example, separately give `mean_radial_resolution`, `minimum_radial_resolution`, `maximum_radial_resolution`, `mean_longitudinal_resolution`, `minimum_longitudinal_resolution`, and `maximum_longitudinal_resolution` instead of combining them into `[mean/minimum/maximum]_[radial/longitudinal]_resolution`?

I meant to include the new versions of the reproject grid definitions.

**Comment by matthewtiscareno** [comment]
> I meant to include the new versions of the reproject grid definitions.

Aren't reprojection_grid_[radial/longitudinal]_sampling_interval the only ones that we want to have, since the grid will be fixed?  Doesn't everything else have to do with the source data product?

**Comment by rfrenchseti** [comment]
> > I meant to include the new versions of the reproject grid definitions.
> 
> Aren't reprojection_grid_[radial/longitudinal]_sampling_interval the only ones that we want to have, since the grid will be fixed? Doesn't everything else have to do with the source data product?

Sorry I think I was just reading the field name wrong. Yes everything is there.

**Comment by rfrenchseti** [comment]
I believe the new definitions are good.

**Comment by rfrenchseti** [comment]
I think the only real constraint on these values is that they're positive (not negative, not zero).

**Comment by matthewtiscareno** [comment]
> I think the only real constraint on these values is that they're positive (not negative, not zero).

I don't think we have generally bothered to set 0 as a minimum value for many of these attributes.  However, we could do that if you prefer it.

@esimpsons3ti was asking about rules such as requiring both minimum and maximum to be set if one of them is set.  She and I can work out a sensible plan regarding this.
